crates/oxc_transformer/src/typescript/enum.rs

```diff
@@ -227,4 +282,263 @@ impl<'a> TypeScript<'a> {
+                // This is a bit tricky because we need to find the BindingIdentifier that corresponds to the identifier reference.
+                // and then we may to evaluate the initializer of the BindingIdentifier.
+                // finally, we can get the value of the identifier and call the `computed_constant_value` function.
+                // See https://github.com/babel/babel/blob/610897a9a96c5e344e77ca9665df7613d2f88358/packages/babel-plugin-transform-typescript/src/enum.ts#L327-L329
+                None
+            }
+            _ => None,
+        }
+    }
+
+    fn evaluate(
+        &self,
+        expr: &Expression<'a>,
+        prev_members: &FxHashMap<Atom<'a>, ConstantValue>,
+    ) -> Option<ConstantValue> {
+        match expr {
+            Expression::Identifier(_) | Expression::MemberExpression(_) => {
+                self.evalaute_ref(expr, prev_members)
+            }
+            Expression::BinaryExpression(expr) => self.eval_binary_expression(expr, prev_members),
+            Expression::UnaryExpression(expr) => self.eval_unary_expression(expr, prev_members),
+            Expression::NumericLiteral(lit) => {
+                Some(ConstantValue::NumericLiteral((lit.raw.to_string(), lit.value)))
+            }
+            Expression::StringLiteral(lit) => Some(ConstantValue::String(lit.value.to_string())),
+            Expression::TemplateLiteral(lit) => {
+                let mut value = String::new();
+                for part in &lit.quasis {
+                    value.push_str(&part.value.raw);
+                }
+                Some(ConstantValue::String(value))
+            }
+            _ => None,
+        }
+    }
+
+    #[allow(clippy::cast_possible_truncation, clippy::cast_precision_loss)]
+    fn eval_binary_expression(
+        &self,
+        expr: &BinaryExpression<'a>,
+        prev_members: &FxHashMap<Atom<'a>, ConstantValue>,
+    ) -> Option<ConstantValue> {
+        let left = self.evaluate(&expr.left, prev_members)?;
+        let right = self.evaluate(&expr.right, prev_members)?;
+
+        if matches!(expr.operator, BinaryOperator::Addition)
+            && (matches!(left, ConstantValue::String(_))
+                || matches!(right, ConstantValue::String(_)))
+        {
+            let left_string = match left {
+                ConstantValue::NumericLiteral((raw, _)) => raw.to_string(),
+                ConstantValue::String(str) => str,
+                ConstantValue::F64(v) => v.to_string(),
+                ConstantValue::Identifier(_) => unreachable!(),
+            };
+
+            let right_string = match right {
+                ConstantValue::NumericLiteral((raw, _)) => raw.to_string(),
+                ConstantValue::String(str) => str,
+                ConstantValue::F64(v) => v.to_string(),
+                ConstantValue::Identifier(_) => unreachable!(),
+            };
+
+            return Some(ConstantValue::String(format!("{left_string}{right_string}")));
```

Comment:
In JavaScript, numbers are converted to strings using a distinct method compared to Rust or C/C++. For instance, the number 1e21 is rendered as `"1e21"` by JavaScript's `toString` method, while Rust would format it as `"1000000000000000000000f"`. 
Similarly, the representation of Infinity varies: JavaScript outputs `"Infinity"` but Rust uses `"inf"`.

Reply:
I mean this line.

https://github.com/oxc-project/oxc/blob/feaf394e4707fd5546bfa42f40e49246700db933/crates/oxc_transformer/src/typescript/enum.rs#L402

Reply:
Thank you, I will add some tests to cover this.